src/__swaps__/screens/Swap/hooks/useSwapOutputQuotesDisabled.ts

```diff
@@ -0,0 +1,27 @@
+import { SharedValue, useAnimatedReaction, useSharedValue } from 'react-native-reanimated';
+
+import { ExtendedAnimatedAssetWithColors } from '@/__swaps__/types/assets';
+
+export const useSwapOutputQuotesDisabled = ({
+  inputAsset,
+  outputAsset,
+}: {
+  inputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
+  outputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
+}) => {
+  const outputQuotesAreDisabled = useSharedValue(false);
+
+  useAnimatedReaction(
+    () => ({
+      inputChainId: inputAsset?.value?.chainId,
+      outputChainId: outputAsset?.value?.chainId,
+    }),
+    current => {
+      if (current.inputChainId && current.outputChainId) {
+        outputQuotesAreDisabled.value = current.inputChainId !== current.outputChainId;
+      }
+    }
+  );
+
+  return outputQuotesAreDisabled;
+};
```

Comment:
Would use a derived value for this:

```suggestion
import { SharedValue, useDerivedValue } from 'react-native-reanimated';
import { ExtendedAnimatedAssetWithColors } from '@/__swaps__/types/assets';

export const useSwapOutputQuotesDisabled = ({
  inputAsset,
  outputAsset,
}: {
  inputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
  outputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
}) => {
  const outputQuotesAreDisabled = useDerivedValue(() => {
    const bothAssetsSelected = !!inputAsset.value && !!outputAsset.value;
    if (!bothAssetsSelected) return false;
    
    return inputAsset.value?.chainId !== outputAsset.value?.chainId;
  });

  return outputQuotesAreDisabled;
};
```